### PR TITLE
UmdTask72_Fall2025_JAX_Wildlife_Image_Classification_2

### PR DESCRIPTION
**Pull request by ravivignesh1999**
This PR is the full implementation of the project "Wildlife Image Classification using JAX"

Name : Ravi Vignesh LNU
UID: 121322302

Issue number: 72

Overview:                                                                
                                                                           
  - Loads the Animals‑10 dataset and preprocesses RGB frames to 128×128    
    tensors.                                                               
  - Implements a configurable JAX/Flax CNN (SimpleCNN) with tunable kernel 
    sizes, dropout, and learning rate.                                     
  - Provides two notebooks:                                                
      - JAX_wildlife.API.ipynb – API tutorial (data ingestion → quick      
        train/eval).                                                       
      - JAX_wildlife.example.ipynb – full workflow with hyper-parameter    
        sweeps and diagnostics.                                            
  - Dockerfile delivers a reproducible CPU/GPU environment.                
                                                                           
  Implementation Highlights:                                               
                                                                           
  - JAX_wildlife_utils.py centralizes data I/O, model definition, training 
    loop, evaluation, and plotting helpers.                                
  - Example notebook saves all diagnostics under outputs/evaluation/       
    (confusion matrix, test metrics, misclassifications, correct examples) 
    and outputs/hparam/ (accuracy plot + per-config confusion matrices).   
  - Hyper-parameter tuning (learning rate, kernels, dropout).              
                                                                           
  Deliverables:                                                            

  1. JAX_wildlife_utils.py – reusable API module.                          
  2. JAX_wildlife.API.ipynb/.md – tutorial.                                
  3. JAX_wildlife.example.ipynb/.md – full end-to-end notebook +           
     documentation.                                                        
  4. Dockerfile – reproducible environment.                                
  5. Organized visualization      
                                                                           
  Getting Started:                                                         
                                                                                      
 - cd TutorTask72_Fall2025_JAX_Wildlife_Image_Classification                
 -  docker build -t msml610/tutortask72_jax .                                
 - docker run --rm -it -p 8888:8888 -v "${PWD}:/data" msml610/              
  tutortask72_jax \                                                        
    bash -lc "cd /data && jupyter-notebook --port=8888 --no-browser        
  --ip=0.0.0.0 --allow-root --NotebookApp.token=''"                        
                                                                           
  Then open http://localhost:8888, run both notebooks via “Restart &       
  Run All,” and review the generated outputs in outputs/evaluation/ and    
  outputs/hparam/.

### CODE REVIEW

**Comment by ravivignesh1999** [comment]
Hi @aver81. I've updated the API files. Pls review my code. Thanks
